src/main/java/com/uid2/admin/vertx/service/EncryptionKeyService.java

```diff
@@ -548,6 +548,15 @@ private int getOrCreateKeysetId(int siteId)
         Keyset keyset = lookUpKeyset(siteId, currentKeysets);
         if(keyset == null) {
             int newKeysetId = getMaxKeyset(currentKeysets)+1;
+            if(siteId == Const.Data.MasterKeySiteId) {
+                newKeysetId = Const.Data.MasterKeysetId;
+            }
+            if(siteId == Const.Data.RefreshKeySiteId) {
```

Comment:
'else-if' is better here as only 1 of 3 cases will match. no need to compare 3 times.

Reply:
fixed